earthkit/data/readers/csv.py

```diff
@@ -134,13 +134,26 @@ def __init__(self, source, path, compression=None):
         self.compression = compression
         self.dialect, self.has_header = probe_csv(path, compression=compression)
 
-    def to_pandas(self, pandas_read_csv_kwargs=None):
+    def to_pandas(
+        self, comment="#", compression=None, pandas_read_csv_kwargs=None, **kwargs
+    ):
         """Convert CSV data into a :py:class:`pandas.DataFrame` using :py:func:`pandas.read_csv`.
 
         Parameters
         ----------
+        comment: str
+            Character that represents a comment line in csv file. This value is ignored if the comment
+            character is defined in pandas_read_csv_kwargs.
+        compression: (None, str)
+            The compression of the file being opened. This value is ignored if Earthkit detects
```

Comment:
The user will not know if earthkit detected the compression, so this decision is hidden.

Reply:
I agree, but this partly why I included it in the doc-string, so that it is clear that Earthkit is trying to do some compression detection. TBH, it probably makes most sense to remove this special treatment we give to compression as I think most out of the box cases the Earthkit detection is doing it all. I was just trying to be as compatible with the existing code.
... maybe this is another point to discuss in person?

Reply:
I have removed compression from the kwargs as I think it is misleading/confusing, earhtkit will overwrite any provided value with what it detects.

I have added a note in the docstring explaining the EK will handle the compreesion of archived csv files